File: backend/src/main/java/com/medistock/purchase/service/PurchaseOrderService.java
package com.medistock.purchase.service;

import com.medistock.purchase.dto.request.PurchaseOrderRequest;
import com.medistock.purchase.dto.response.PurchaseOrderResponse;
import com.medistock.purchase.dto.request.PurchaseOrderReceiptRequest;
import java.util.List;

public interface PurchaseOrderService {
    PurchaseOrderResponse createPurchaseOrder(PurchaseOrderRequest request);
    List<PurchaseOrderResponse> getAllPurchaseOrders();
    PurchaseOrderResponse getPurchaseOrderById(Long id);
    PurchaseOrderResponse updatePurchaseOrderStatus(Long id, String status);
    PurchaseOrderResponse updatePurchaseOrderStatus(Long id, String status, String note);
    void deletePurchaseOrder(Long id);
        PurchaseOrderResponse receivePurchaseOrder(
            Long id,
            PurchaseOrderReceiptRequest request
    );
}

File: frontend/src/services/api/purchaseOrderService.js
import apiClient from './apiClient';

export const purchaseOrderService = {
  getAllPurchaseOrders: async () => {
    const response = await apiClient.get('/purchase-orders');
    return response.data;
  },

  getPurchaseOrderById: async (id) => {
    const response = await apiClient.get(`/purchase-orders/${id}`);
    return response.data;
  },

  createPurchaseOrder: async (data) => {
    const response = await apiClient.post('/purchase-orders', data);
    return response.data;
  },

  updatePurchaseOrderStatus: async (id, status) => {
    const response = await apiClient.patch(`/purchase-orders/${id}/status`, { status });
    return response.data;
  },

  deletePurchaseOrder: async (id) => {
    const response = await apiClient.delete(`/purchase-orders/${id}`);
    return response.data;
  },

  receivePurchaseOrder: async (id, receiptData) => {
  const response = await apiClient.post(
    `/purchase-orders/${id}/receive`,
    receiptData
  );
  return response.data;
},
};
